### PR TITLE
chore: use self-created 1es hosted pools to build windows vhd

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
Use self-created 1es hosted pools to build windows vhd so that we do not need to create security exception to expose WinRM port 5986 in the sub “AKS VHD Image Release”

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by cameronmeissner — APPROVED** [review]
I just double-checked and looks like you did this already, but I'll just stress here to make sure the names of your pools are unique and don't overlap with existing Linux pools (I think the bicep templates would probably yell at you for that anyways now that I think about it).

**Comment by cameronmeissner** [comment]
I'll also mention that we've been experiencing network issues with the pool and connecting to VMs in the peered vnet during our ARM64 builds specifically - I doubt this will be something that affects you but we're planning on getting around to fixing that on our side soon.

File: vhdbuilder/packer/windows-vhd-builder.json [hunk 1]
```diff
@@ -49,10 +50,9 @@
                 "createdBy": "aks-vhd-pipeline",
                 "SkipASMAzSecPack": "True"
             },
-            "virtual_network_resource_group_name": "{{user `vnet_resource_group_name`}}",
-            "virtual_network_name": "vnet",
-            "virtual_network_subnet_name": "subnet",
-            "private_virtual_network_with_public_ip": "true"
+            "virtual_network_resource_group_name": "nodesigtest-agent-pool",
```
Comment (ShiqianTao) [inline vhdbuilder/packer/windows-vhd-builder.json:53]: Variable may be more flexible. BTW: this work is interesting, could we have a wiki to note it?

Comment (junjiezhang1997) [reply vhdbuilder/packer/windows-vhd-builder.json]: Fine. Modify the use of variable and add the wiki.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**